src/Data/Replicator/Linear/Internal.hs

```diff
@@ -133,13 +133,26 @@ extend f = map f . duplicate
 --
 -- > elim (,) :: Replicator a %1 -> (a, a)
 -- > elim (,,) :: Replicator a %1 -> (a, a, a)
-elim :: forall (n :: Nat) a b f. (Elim (NatToPeano n) a b, IsFunN a b f, f ~ FunN (NatToPeano n) a b, n ~ Arity b f) => f %1 -> Replicator a %1 -> b
+elim ::
+  forall (n :: Nat) a b f.
+  ( -- Always fulfilled. Provides the actual implementation of 'elim' in
+    -- terms of @elim\'@
+    Elim (NatToPeano n) a b,
+    -- Asserts that @f@ is a function taking @n@ linear arguments of type
+    -- @a@ and returning a value of type @b@
+    IsFunN a b f,
+    f ~ FunN (NatToPeano n) a b,
+    n ~ Arity b f
+  ) =>
+  f %1 ->
+  Replicator a %1 ->
+  b
 elim f r = elim' @(NatToPeano n) f r
 
--- | @'Elim' n a b f@ asserts that @f@ is a function taking @n@ linear arguments
--- of type @a@ and then returning a value of type @b@.
+-- | @'Elim' n a b@ is used to avoid recursion in the implementation of 'elim'
+-- so that 'elim' can be inlined.
```

Comment:
```suggestion
-- | @'Elim' n a b@ is used to implement 'elim' without recursion
-- so that we can guarantee that 'elim' will be inlined and unrolled.
```